var/spack/repos/builtin/packages/samrai/package.py

```diff
@@ -53,24 +55,26 @@ class Samrai(Package):
     depends_on("zlib")
     depends_on("hdf5+mpi")
     depends_on("boost")
+    depends_on("m4")
 
     # don't build tools with gcc
     patch('no-tool-build.patch', when='%gcc')
 
-    # TODO: currently hard-coded to use openmpi - be careful!
     def install(self, spec, prefix):
         configure(
             "--prefix=%s" % prefix,
-            "--with-CXX=%s" % spec['mpi'].prefix.bin + "/mpic++",
-            "--with-CC=%s" % spec['mpi'].prefix.bin + "/mpicc",
+            "--with-CXX=%s" % spec['mpi'].mpicxx,
+            "--with-CC=%s" % spec['mpi'].mpicc,
+            "--with-F77=%s" % spec['mpi'].mpifc,
+            "--with-M4=%s" % spec['m4'].prefix,
             "--with-hdf5=%s" % spec['hdf5'].prefix,
             "--with-boost=%s" % spec['boost'].prefix,
             "--with-zlib=%s" % spec['zlib'].prefix,
             "--without-blas",
             "--without-lapack",
             "--with-hypre=no",
             "--with-petsc=no",
-            "--enable-opt",
+            "--enable-opt=-O3",
             "--disable-debug")
```

Comment:
I can not confirm the exact behavior of this flag as it is not listed in the configure help, but it is a flag I have been using for years in SAMRAI, and since it was already in the package file before I modified it, I assume the original author found it fairly standard as well. I would also rather have potential optimizations it may provide, and forego the debug symbols it may provide otherwise.